lib/ansible/module_utils/redfish_utils.py

```diff
@@ -0,0 +1,932 @@
+        response = self.get_request(self.root_uri + bios_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        boot_mode = data[u'Attributes']["BootMode"]
+        if boot_mode == "Uefi":
+            boot_seq = "UefiBootSeq"
+        else:
+            boot_seq = "BootSeq"
+
+        response = self.get_request(self.root_uri + self.systems_uri + "/" + bootsources)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+
+        boot_device_list = data[u'Attributes'][boot_seq]
+        for b in boot_device_list:
+            boot_device = {}
+            boot_device["Index"] = b[u'Index']
+            boot_device["Name"] = b[u'Name']
+            boot_device["Enabled"] = b[u'Enabled']
+            boot_device_details.append(boot_device)
+        result["entries"] = boot_device_details
+        return result
+
+    def set_bios_default_settings(self):
+        result = {}
+        key = "Bios"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        bios_uri = data[key]["@odata.id"]
+
+        # Extract proper URI
+        response = self.get_request(self.root_uri + bios_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        reset_bios_settings_uri = data["Actions"]["#Bios.ResetBios"]["target"]
+
+        response = self.post_request(self.root_uri + reset_bios_settings_uri, {}, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def set_one_time_boot_device(self, bootdevice):
+        result = {}
+        key = "Bios"
+
+        # Search for 'key' entry and extract URI from it
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        result['ret'] = True
+        data = response['data']
+        bios_uri = data[key]["@odata.id"]
+
+        response = self.get_request(self.root_uri + bios_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        boot_mode = data[u'Attributes']["BootMode"]
+        if boot_mode == "Uefi":
+            payload = {"Boot": {"BootSourceOverrideTarget": "UefiTarget", "UefiTargetBootSourceOverride": bootdevice}}
+        else:
+            payload = {"Boot": {"BootSourceOverrideTarget": bootdevice}}
+
+        response = self.patch_request(self.root_uri + self.systems_uri, payload, HEADERS)
+        if response['ret'] is False:
+            return response
+        return {'ret': True}
+
+    def set_manager_attributes(self, attr):
+        result = {}
```

Comment:
Should this be `response` since that is what is returned? I don't see `result` used within this function.